pkg/kapp/config/config.go

```diff
@@ -53,6 +54,18 @@ type WaitRuleConditionMatcher struct {
 	Success bool
 }
 
+type WaitRuleYtt struct {
+	// Contracts are named and versioned (eg v1)
+	// to provide a stable interface to rule authors.
+	// Multiple contracts will be offered at the same time
+	// so that existing rules do not not break as we decide to evolve running environment.
+	WaitRuleContractV1 *WaitRuleYttContractV1 `json:"waitRuleContractV1"`
```

Comment:
waitRule in the name is redundant (since rules are under waitRules). in following example [0], ive used overlayContractV1 because main logic is done via ytt "overlay". should we call this something like funcContractV1 since main logic is done via a function?

[0] https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/config/default.go#L125-L126

Reply:
Adopted. Changed it to `funcContractV1 `